Introducing Conan as our package manager

beckerhe opened the pull request:
> *Important information upfront*: After this change you have to clean your build directories and bootstrap again.

This PR replaces `vcpkg` by `conan`, which is another popular package manager for C and C++.

### Introducing conan
Conan is written in Python3, open source, MIT licensed. It has a decentral approach to package management and binary distribution, meaning you can register as many package repositories (called remotes like in git) as you like.

### Dependencies
I transferred all dependencies that have been in vcpkg to conan. For some of them, I had to make changes to the packaging instructions (called recipes). All these recipes are checked in (in `contrib/conan/recipes`).

### Prebuilts
I created prebuilts (called packages in conan) for all of our current dependencies if not available in one of the two official remotes. The following build configurations (called profiles in conan) are available:

* Linux clang{7,8,9} {release, debug}
* Linux gcc{8,9} {release, debug}
* GGP Linux clang7 {release, debug}
* Windows MSVC{2017,2019} {release,debug} {x86,x64}

In total 20 build configurations. I'm thinking about adding RelWithDebInfo packages. Please let me know, what you think about that.

The packages are currently hosted on Bintray (Bintray is free for public remotes containing only open-source packages, which is true in our case.) We should discuss in a later step, what we be the long term solution.

### Changes to the build scripts

While doing this I tried to unify the build and bootstrap scripts. There is now a `bootstrap-orbit.{sh,bat}` and a `build.{sh,bat}` script. The first one is checking for system dependencies (including conan itself), tries to install them and also installs some conan configuration files. Afterwards it calls `build.{sh,bat}` which is taking care of the build. By calling `build.{sh,bat}` without any arguments it's going to build the default configurations. On Linux that the default system compiler in debug and release configuration. On Windows it is 32bit and 64bit release build with the default compiler (which is the most recent MSVC.

There was some discussion about changing the default build profiles (Building clang7_debug and clang7_release on Linux by default). But for now this PR is not changing that behaviour.

Apart from calling `build.{sh,bat}` without any arguments you can also supply a list of conan profiles (Check out `contrib/conan/config/profiles` for the ones provided by this PR.) For example calling
```bash
./build.sh clang7_release gcc9_debug clang8_release
```
will build the three mentioned profiles, instead of the default ones.

*Side Note:* You can also provide build configurations as arguments to the boostrap scripts and they will be forwarded to the build script.

*Directory structure:* I know, there has been some discussion about simplifying the build directory structure. The `build.{sh,bat}`-script currently creates a subdirectory called `build_<profile_name>` in the current working directory. Moving over to different schemes like `build/<profile_name>` is easy and can be done at a later stage.

ronaldfw commented on bootstrap-orbit.sh:
```diff
@@ -1,61 +1,46 @@
+
+  which conan >/dev/null
+  if [ $? -ne 0 ]; then
+    echo "Could not find conan in the path, although the installation report success."
```
```suggestion
    echo "Could not find conan in the path, although the installation reported success."
```

ronaldfw commented on build.sh:
```diff
@@ -0,0 +1,43 @@
+  return $?
+}
+
+
```
Nit: Remove one empty line.

beckerhe commented:
> Thanks for this!
> 
> Added some small comments, overall looks good.
> 
> The main concern I have here is how convinced we are that this is going to be our long-term solution. If there is still doubt and open questions, then this (and every time we change anything around building and dependencies) adds a lot of churn that would be good to avoid. If you think it will be, then we should go this way and add what is needed. (My understanding is the latter.)

Hi Ronald,
thanks for your feedback. Muhc appreciated. I will get right on to make the changes you suggested.

Regarding your main concern: Yes, that's basically the question I was asking myself the whole time. Is it worh doing that? There is one thing I am pretty sure about: We will get nowhere near a reliable prebuilt package distribution solution based on vcpkg. There are so many subtleties involved, when you wanna deploy binary packages and we are not even talking about cross compilation. Conan handles these for you, while vcpkg does not even have concepts for this.

So I see it like this: If we want to have prebuilt binaries for our dependencies, there is no way around conan. If building from source is fine, this change might not be worth it.

beckerhe commented on DEVELOPMENT.md:
```diff
@@ -1,50 +1,36 @@
-the `bootstrap-orbit.sh` file on how it is done.
+There are some exceptions. On Linux, we rely by default on the distribution's Qt5
+and Mesa installation. This can be changed by modifying the conan package options
+`system_qt` and `system_mesa`.
```
Better now?

ronaldfw commented on DEVELOPMENT.md:
````diff
@@ -1,51 +1,43 @@
-which cmake # Should point to ~/.local/bin/cmake
-cmake --version # Should output 3.15.2 or higher
-```
+To build Orbit you need a compiler, capable of C++17. The following ones should be fine.
````
```suggestion
To build Orbit you need a compiler capable of C++17. The following ones should be fine.
```

ronaldfw commented:
> So I see it like this: If we want to have prebuilt binaries for our dependencies, there is no way around conan. If building from source is fine, this change might not be worth it.

Ok, based on what you said, this is for sure strictly better than vcpkg for our purposes. So let's go with this unless we find insurmountable issues.

beckerhe commented on DEVELOPMENT.md:
````diff
@@ -79,34 +71,25 @@ openssl.
 sudo apt install libglu1-mesa libxi6 libxmu6
 ```
+
+before the service can be started:
````
This could have never been working. I fixed the instructions. Thanks for trying that!

beckerhe commented:
> @beckerhe , thanks for this _monumental_ change, compiling from scratch is insanely fast now, orders of magnitude faster than compiling everything through vcpkg. Amazing work.
> 
> I tested it on gamelet and everything runs fine. I also compiled and launched on Windows, everything looks good.
> 
> I did however run into small issues:
> 
> Linux:
> 
> * had to manually install python3-pip (sudo apt-get install python3-pip), this could potentially be done automatically in the bootstrap scripts.
> * after installing Conan, it wasn't available on the command line, I had to add _$HOME/.local/bin_ in my PATH.  This should be added to the docs.
I added python3-pip to the apt install call and wrote some words about adding ./local/bin to the PATH.
> 
> GGP
> 
> * had to install the Google Cloud SDK to fix this error: _ConanException: Error 127 while executing gsutil cp gs://..._
> * had to run _gcloud auth login_
> * Exact steps could be added to internal docs...
That's already on my todo-list. I will updated the internal docs as soon as this is merged.
> 
> Windows
> 
> * Had to install Python3 AND manually update my environment variables so that the new version would show up before the previous install (2.7) in the PATH variable.  This could be mentioned in the doc.

I added some words about this to the DEVELOPMENT.md file. I'm not entirely sure what's the
problem with the python-3 and the python-2.7 installation. These should not affect each other. We should discuss this.
> 
> Otherwise, again, thank you for this great PR. This is extremely useful as it drastically reduces compile times and sets the table for proper packaging. Thank you!
